### PR TITLE
modules: add support to build Xtensa HAL as a module

### PR DESCRIPTION
**Pull request by dcpleung**
Requires: https://github.com/zephyrproject-rtos/hal_xtensa/pull/1

This adds initial support for building Xtensa HAL as a Zephyr module.

### CODE REVIEW

**Review by SebastianBoe — APPROVED** [review]

**Review by andyross — APPROVED** [review]

File: west.yml [hunk 1]
```diff
@@ -103,6 +103,9 @@ manifest:
     - name: mipi-sys-t
       path: modules/debug/mipi-sys-t
       revision: baf51863f19f009b92e762115ba5572a5b996b92
+    - name: hal_xtensa
+      revision: d4c49675b5b49e66ae69fe8b90d5f5290c63ca2e
```
Comment (nashif) [inline west.yml]: dont you need to be pointing to zephyrproject-rtos/hal_xtensa#1?

**Review by nashif — CHANGES_REQUESTED** [review]

File: west.yml (hunk 1 above)
Comment (dcpleung) [reply west.yml]: This should be as https://github.com/zephyrproject-rtos/hal_xtensa/pull/1/commits/d4c49675b5b49e66ae69fe8b90d5f5290c63ca2e is the latest commit there. Once the HAL PR is merged, I will double-check the SHA.

Comment (nashif) [reply west.yml]: You can do revision: pull/1/head, so this PR is testing with the HAL PR.

Comment (dcpleung) [reply west.yml]: Updated west.yml so CI can run. Will revert once the HAL PR is merged.

Comment (dcpleung) [reply west.yml]: HAL PR is merged, and the revision has been updated to latest commit from there.

**Review by nashif — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**